Automattic/Automattic-Tracks-iOS
# Add device_info_app_build_configuration property to all events

Adds a new `device_info_app_build_configuration` property to all events that can be:
- Debug: for local developer builds
- Alpha: for `Release-alpha` builds, I believe we use this for Installable Builds on PRs. This requires the app integrating the library to define an `ALPHA` preprocessor macro.
- Beta: for `Release` builds uploaded to TestFlight
- Production: for `Release` builds uploaded to the App Store

Alpha releases can be configured with the following in the app's `Podfile`:
```ruby
  # Flag Alpha builds for Tracks
  # ============================
  installer.pods_project.targets.each do |target|
    next unless target.name == "Automattic-Tracks-iOS"
    target.build_configurations.each do |config|
      next unless config.name == "Release-Alpha"
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', 'ALPHA=1']
    end
  end
```

I've only implemented this for Woo in https://github.com/woocommerce/woocommerce-ios/pull/4468/, so I'm not sure if integrating this in WordPress will require any changes for the WP Internal builds.

Part of https://github.com/woocommerce/woocommerce-ios/issues/4432

## To test

1. Check out https://github.com/woocommerce/woocommerce-ios/pull/4468 and run the Woo app.
2. You can set a breakpoint in `- [TracksServiceRemote sendBatchOfEvents:withSharedProperties:completionHandler:]` and ensure that the `properties` argument contains `Debug` for `device_info_app_build_configuration`.
3. Check the Tracks Live View after a few minutes, and verify that the events coming from the app include the new property.

I haven't been able to test anything other than Debug builds yet 😞
